### DISCUSSION TITLE
Does deprecated syntax need a time guarantee of continual functionality?

### DISCUSSION THREAD

**Discussion by Wajinn** [vlang/v#16313] [General]
To clarify what I mean by the title, is that recently Vlang switch to `{var}`, instead of using `$var` and `${var}` for string interpolation.  Great, maybe this can be argued as an improvement, if `'{{{var}}}'` is not considered visually confusing.  However, lots of code has the older syntax, `${var}`, which needs to be replaced if it isn't going to be kept (though perhaps it should be kept around).

When these syntax changes are made, perhaps an estimated expiration date and warning should be given.  The warning should not prevent the code from running or compiling, and have the estimated date of when that syntax will no longer work.  Sometime after the expiration date, usage of such syntax becomes an error.  What say ye all?

**Comment by medvednikov** [comment]
Yes, that's the strategy we've been using.

It'll work for 1-2 years, and will result in a deprecation warning.

vfmt will automatically update the code.

**Reply by Wajinn** [reply]
Thanks for the clarification.
